fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -200,10 +200,20 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
                 override fun afterTextChanged(editable: Editable?) {
                     // update usernameChanged to true when the text is not empty,
                     // otherwise it is not changed, as this screen starts with an empty username.
-                    usernameChanged = editable.toString().isNotEmpty()
-                    updateUsernameField()
-                    setSaveButtonState()
-                    findDuplicate()
+                    when {
+                        editable.toString().isEmpty() -> {
+                            validUsername = false
+                            binding.clearUsernameTextButton.isVisible = false
+                        }
+                        else -> {
+                            usernameChanged = editable.toString().isNotEmpty()
+                            binding.clearUsernameTextButton.isVisible = !editable.isNullOrBlank()
```

Comment:
Here we are using a check on whether the text is `blank` but above only if the text is `empty`. Based on the current code it looks like we should stick with a check for just `empty`?

Reply:
Thanks!
But it seems that the clear button does not allows for only spaces while other functionality does - because elsewhere we only do `isEmpty` checks?

Reply:
@Mugurell `other functionality does` what does it mean? Do you think we need to remove blank space check at beginning of?

Reply:
I would expect to use only one check: `isBlank()` or `isEmpty()` everywhere not a combination of them.
At line 209  `isNotEmpty` is used and then at line 210 `isNullOrBlank` is used.
Looking at the current code it seems like we are not checking for `blank`.
And checking with desktop I see that is allows for usernames or passwords composed solely from empty spaces so I recommend here also to only check whether the string is empty or not and avoid the `isBlank` check.